### ISSUE TITLE
Field to store visit occurrence source key

### ISSUE THREAD

**Issue by MaximMoinat** [OHDSI/Themis#205]
I see a need for storing the original key of the visit/encounter from the source tables in the OMOP visit_occurrence table. This is helpful for a simple look up of the visit_occurrence_id when linking events to the visit. I propose the field `visit_occurrence_source_value`, with a description similar to person_source_value: "Use this field to link back to visits in the source data. This is typically used for lookups in the ETL implementation.".

Now you might say, one can add their own local fields or lookup table for this. Which is true, but what I see now is that a lot of OMOP instances repurpose the the `visit_source_value` to store the source key of the visit. This is somewhat inconsistent, as the `visit_source_value` should be used for storing the source code linked to the `visit_concept_id`. 

```mermaid
erDiagram
  VISIT_OCCURRENCE {
    visit_occurrence_id integer
    VISIT_OCCURRENCE_SOURCE_VALUE integer
    person_id integer
    visit_concept_id integer
    visit_start_date date
    visit_start_datetime datetime
    visit_end_date date
    visit_end_datetime datetime
    visit_type_concept_id Integer
    provider_id integer
    care_site_id integer
    visit_source_value varchar
    visit_source_concept_id integer
    admitted_from_concept_id integer
    admitted_from_source_value varchar
    discharged_to_concept_id integer
    discharged_to_source_value varchar
    preceding_visit_occurrence_id integer
  }
```

**Comment by clairblacketer** [comment]
@MaximMoinat after reviewing, I believe this would be better served as a site specific add-on. If you agree, I will close this.

**Comment by MaximMoinat** [comment]
@clairblacketer I still think this would be a simple and very helpful new field. But if we think this should be site-specific, I would like to transfer this issue to Themis to write a short convvention on how to store the original key of the visit. Many people are not aware that they can add custom fields for their local prupose.

Clair, could you Transfer this issue to the Themis repo? (see bottomleft, -> Transfer issue) I do not have the right priviliges for that

**Comment by clairblacketer** [comment]
@MaximMoinat we can still discuss this at the CDM WG meeting at the symposium if you like and get the community's take on it. When speaking with @MelaniePhilofsky yesterday the concern was that groups may push for other ids like this in other tables. I am happy to keep it open for discussion until after the 2025 meeting

**Comment by MelaniePhilofsky** [comment]
I agree, @MaximMoinat, we need to write a Themis convention for adding additional fields and tables to the OMOP CDM. Please see [this poster](https://www.ohdsi.org/2021-global-symposium-showcase-18/) I presented at the 2021 OHDSI Global Symposium. It specifically covers adding extra columns to store source keys enabling linkage back to the original record. It also covers when to add extension columns to your local CDM and when to ask the CDM WG to add additional columns to the canonical CDM.

**Comment by clairblacketer** [comment]
@MaximMoinat @MelaniePhilofsky we have this [documentation](https://ohdsi.github.io/CommonDataModel/typesOfAdditions.html) describing each type of add-on and when to seek additions to the model itself which I think we could update based on the poster. In this case I think the convention would be related to storing the source key of a visit in as a site-specific field

**Comment by MelaniePhilofsky** [comment]
Great! We can create a general Themis convention and point folks to the documentation you link above.

**Comment by MelaniePhilofsky** [comment]
We have documentation for site-specific add-ons. We need to review it for clarity, add an example or two and bring it over to the Themis Convention Library.
